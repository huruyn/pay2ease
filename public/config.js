window.API_URL = "https://zippy-retail-api-dev.azurewebsites.net";