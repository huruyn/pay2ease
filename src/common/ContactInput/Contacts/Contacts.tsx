import PropTypes from 'prop-types';
import { Button, FormControl } from 'react-bootstrap';
import {
  Dispatch,
  SetStateAction,
  useEffect,
  useState,
} from 'react';
import { Contact } from '../../../api';
import { ContactTypeEnum } from '../../../constants/enum/ContactTypeEnum';
import { formatContactName } from '../../../Helpers';
import AddContact from '../AddContact/AddContact';

function Contacts({
  contactList,
  setContactList,
  selectedContact,
  setSelectedContact,
  showContactList,
  setShowContactList,
}: {
  contactList: Array<Contact>;
  setContactList: Dispatch<SetStateAction<Array<Contact>>>;
  selectedContact: Contact;
  setSelectedContact: Dispatch<SetStateAction<Contact>>;
  showContactList: boolean,
  setShowContactList: Dispatch<SetStateAction<boolean>>;
}) {
  const [contactType, setContactType] = useState<ContactTypeEnum>(ContactTypeEnum.PERSONAL);
  const [isProcessing, setIsProcessing] = useState<boolean>(true);
  const [filteredContactList, setFilteredContactList] = useState<Array<Contact>>(contactList);
  const [searchValue, setSearchValue] = useState<string>('');
  const [initialized, setInitialized] = useState<boolean>(false);
  const [isNewContact, setIsNewContact] = useState<boolean>(false);

  const filter = (value: string) => {
    setSearchValue(value);
    setFilteredContactList(contactList.filter((contactItem) => {
      const search = value.toLowerCase();
      return contactItem?.firstName?.toLowerCase().includes(search)
        || contactItem?.lastName?.toLowerCase().includes(search)
        || contactItem?.email?.toLowerCase().includes(search);
    }));
  };

  const contactCreate = (contact: Contact) => {
    const newContactAdded = [...contactList, contact];
    setContactList(newContactAdded);
    setIsNewContact(false);
    setShowContactList(true);
    setInitialized(false);
  };

  useEffect(() => {
    if (!initialized && contactList.length) {
      setInitialized(true);
      setIsProcessing(false);
      filter('');
    }

    if (showContactList) {
      setIsNewContact(false);
    }
  });

  return (
    <>
      <div className={`contact-container ${!isNewContact && showContactList ? 'visible' : 'hidden'}`}>
        <div className="zippy-btn-group btn-group" role="group" aria-label="Zippy Cash">
          <Button
            className={contactType === ContactTypeEnum.PERSONAL ? 'active' : ''}
            onClick={() => setContactType(ContactTypeEnum.PERSONAL)}
            disabled={contactType === ContactTypeEnum.PERSONAL}
          >
            Personal
          </Button>
          <Button
            className={contactType === ContactTypeEnum.BUSINESS ? 'active' : ''}
            onClick={() => setContactType(ContactTypeEnum.BUSINESS)}
            disabled={contactType === ContactTypeEnum.BUSINESS && true}
          >
            Business
          </Button>
        </div>
        <div className="contact-search">
          <i className="zippy-cash-icon zc-search" />
          <FormControl
            placeholder="Search..."
            value={searchValue}
            onChange={(evt) => filter(evt.target.value)}
          />
        </div>
        <div className="contact-list-container">
          {filteredContactList.length > 0
          && (
          <div className="personal-contacts">
            <h4>Contacts</h4>
            {filteredContactList?.map((item: Contact) => (
              <div className={`contact-item${selectedContact.id === item.id ? ' selected' : ''}`} key={item.id}>
                <div className="contact-name">
                  {formatContactName(item.firstName, item.lastName)}
                </div>
                <button
                  className="btn-select-contact btn-transparent"
                  type="button"
                  onClick={() => {
                    if (item.email) {
                      setSelectedContact(item);
                      setShowContactList(false);
                    }
                  }}
                >
                  <i className="zippy-cash-icon zc-personal" />
                </button>
              </div>
            ))}
          </div>
          )}
          {!initialized && filteredContactList.length === 0 && searchValue === '' && (
          <div>Loading...</div>
          )}
          {initialized && filteredContactList.length === 0 && searchValue !== '' && (
          <div>No match</div>
          )}
          {initialized && filteredContactList.length === 0 && searchValue === '' && (
          <div>No contacts</div>
          )}
        </div>
        <div className="action">
          <Button
            className="zippy-btn"
            disabled={isProcessing}
            onClick={() => {
              setShowContactList(false);
              setIsNewContact(true);
            }}
          >
            {isProcessing && <div className="loading spinner-border" role="status" />}
            Add Contact
          </Button>
        </div>
      </div>
      <div className={`contact-container ${isNewContact && !showContactList ? 'visible' : 'hidden'}`}>
        <AddContact
          onContactCreated={(contact: Contact) => contactCreate(contact)}
        />
      </div>
    </>
  );
}

Contacts.propTypes = {
  contactList: PropTypes.arrayOf(PropTypes.object),
  setContactList: PropTypes.func,
  selectedContact: PropTypes.objectOf(PropTypes.any),
  setSelectedContact: PropTypes.func,
  showContactList: PropTypes.bool,
  setShowContactList: PropTypes.func,
};

Contacts.defaultProps = {
  contactList: [],
  setContactList: undefined,
  selectedContact: {
    id: null,
    email: '',
    firstName: '',
    lastName: '',
    phone: '',
  },
  setSelectedContact: undefined,
  showContactList: false,
  setShowContactList: undefined,
};

export default Contacts;
