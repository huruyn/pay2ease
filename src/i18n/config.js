import i18n from 'i18next';
import { initReactI18next } from 'react-i18next';

i18n.use(initReactI18next).init({
  fallbackLng: 'en-US',
  lng: 'en-US',
  resources: {
    'en-CA': {
      translations: require('./locales/en-CA/translations.json')
    },
    'en-US': {
      translations: require('./locales/en-US/translations.json')
    },
    'es-US': {
      translations: require('./locales/es-US/translations.json')
    },
    'fr-CA': {
      translations: require('./locales/fr-CA/translations.json')
    },
  },
  ns: ['translations'],
  defaultNS: 'translations'
});

i18n.languages = ['en-CA', 'en-US', 'es-US', 'fr-CA'];

export default i18n;