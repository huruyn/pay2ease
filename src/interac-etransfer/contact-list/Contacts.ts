const CONTACTS = [
  {
    key: 1,
    name: '392 Jones',
    lang: 'English',
    email: 'lu_ben2002@yahoo.com',
    phone: '',
  },
  {
    key: 2,
    name: 'Auto Island Inc',
    lang: 'English',
    email: '',
    phone: '647-747-6109',
  },
  {
    key: 3,
    name: 'BITVO',
    lang: 'English',
    email: '42000023186@bitvodeposits.com',
    phone: '',
  },
  {
    key: 4,
    name: 'BITVO.COM',
    lang: 'English',
    email: 'noreply@bitvo.com',
    phone: '',
  },
  {
    key: 5,
    name: 'Chelsea Tough',
    lang: 'English',
    email: 'chelsea.tough@gmail.com',
    phone: '',
  },
  {
    key: 6,
    name: 'Christian',
    lang: 'English',
    email: 'pratt540@gmail.com',
    phone: '',
  },
  {
    key: 7,
    name: 'DnF',
    lang: 'English',
    email: 'franjessco@sympatico.ca',
    phone: '',
  },
  {
    key: 8,
    name: 'DUANE TOUGH',
    lang: 'English',
    email: 'dtough@hotmail.com',
    phone: '647-857-6694',
  },
  {
    key: 9,
    name: 'Gigadat',
    lang: 'English',
    email: 'etransfer@orderdeposit.com',
    phone: '',
  },
  {
    key: 10,
    name: 'J Grundy',
    lang: 'English',
    email: 'jegrundy@rogers.com',
    phone: '',
  },
  {
    key: 11,
    name: 'JJJ',
    lang: 'English',
    email: 'toughj@telus.net',
    phone: '',
  },
  {
    key: 12,
    name: 'Jon English',
    lang: 'English',
    email: 'john.oliverscoffee@gmail.com',
    phone: '',
  },
  {
    key: 13,
    name: 'Karen',
    lang: 'English',
    email: 'english_karen@yahoo.com',
    phone: '',
  },
  {
    key: 14,
    name: 'Mathew',
    lang: 'English',
    email: 'mathewm1212@gmail.com',
    phone: '',
  },
  {
    key: 15,
    name: 'Michael Sovereign Medical',
    lang: 'English',
    email: 'hello@sovereigncosmeticsurgery.com',
    phone: '',
  },
  {
    key: 16,
    name: 'Rishi',
    lang: 'English',
    email: 'rishi.jokhu@gmail.com',
    phone: '',
  },
  {
    key: 17,
    name: 'Robb MacDonald',
    lang: 'English',
    email: 'rm@lawintoronto.com',
    phone: '',
  },
  {
    key: 18,
    name: 'RobnGeo',
    lang: 'English',
    email: 'robngeo@gmail.com',
    phone: '',
  },
];

export default CONTACTS;
