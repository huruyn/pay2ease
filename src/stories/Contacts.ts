import { Contact } from '../api';

const CONTACTS : Contact[] = [
  {
    id: 1,
    firstName: 'Jon',
    lastName: 'Doe',
    phone: '647-747-6109',
    email: 'email1@zipy.cash',
  },
  {
    id: 2,
    firstName: 'Abram',
    lastName: 'Acotas',
    phone: '',
    email: 'email2@zipy.cash',
  },
  {
    id: 3,
    firstName: 'Ragnar',
    lastName: 'Hagan',
    phone: '647-747-3333',
    email: '',
  },
  {
    id: 4,
    firstName: 'Bob',
    lastName: 'Mathew',
    phone: '647-747-4444',
    email: 'email4@zipy.cash',
  },
  {
    id: 5,
    firstName: 'Rafi',
    lastName: 'Doe',
    phone: '647-747-5555',
    email: 'email5@zipy.cash',
  },
];

export default CONTACTS;
