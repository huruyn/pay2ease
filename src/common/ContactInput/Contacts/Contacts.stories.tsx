import { ComponentStory, ComponentMeta } from '@storybook/react';
import '../../../i18n/config';
import '../../../styles/index.scss';
import Contacts from './Contacts';

export default {
  title: 'common/ContactInput',
  component: Contacts,
} as ComponentMeta<typeof Contacts>;

const Template: ComponentStory<typeof Contacts> = () => <Contacts />;

export const Default = Template.bind({});
Default.args = {};
