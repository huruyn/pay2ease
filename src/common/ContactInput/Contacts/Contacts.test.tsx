import '../../../i18n/config';
import Contacts from './Contacts';

const ReactTestRenderer = require('react-test-renderer');

describe('Contacts Component', () => {
  it('matches the snapshot', () => {
    const tree = ReactTestRenderer.create(
      <Contacts />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
