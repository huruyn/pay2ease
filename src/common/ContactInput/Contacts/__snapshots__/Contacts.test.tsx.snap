// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contacts Component matches the snapshot 1`] = `
Array [
  <div
    className="contact-container hidden"
  >
    <div
      aria-label="Zippy Cash"
      className="zippy-btn-group btn-group"
      role="group"
    >
      <button
        className="active btn btn-primary"
        disabled={true}
        onClick={[Function]}
        type="button"
      >
        Personal
      </button>
      <button
        className="btn btn-primary"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        Business
      </button>
    </div>
    <div
      className="contact-search"
    >
      <i
        className="zippy-cash-icon zc-search"
      />
      <input
        className="form-control"
        onChange={[Function]}
        placeholder="Search..."
        value=""
      />
    </div>
    <div
      className="contact-list-container"
    >
      <div>
        Loading...
      </div>
    </div>
    <div
      className="action"
    >
      <button
        className="zippy-btn btn btn-primary"
        disabled={true}
        onClick={[Function]}
        type="button"
      >
        <div
          className="loading spinner-border"
          role="status"
        />
        Add Contact
      </button>
    </div>
  </div>,
  <div
    className="contact-container hidden"
  >
    <div
      className="container-header"
    >
      <h2
        className="title"
      >
        Add  Contact
      </h2>
    </div>
    <div
      className="container-body"
    >
      <input
        className="common-input form-control"
        name="firstname"
        onChange={[Function]}
        placeholder="Firstname"
        type="text"
        value=""
      />
      <input
        className="common-input form-control"
        name="lastname"
        onChange={[Function]}
        placeholder="Lastname"
        type="text"
        value=""
      />
      <input
        className="common-input form-control"
        name="email"
        onChange={[Function]}
        placeholder="Email"
        type="text"
        value=""
      />
      <input
        className="common-input form-control"
        name="nickname"
        onChange={[Function]}
        placeholder="Nickname (optional)"
        type="text"
        value=""
      />
    </div>
    <div
      className="action"
    >
      <button
        className="zippy-btn btn btn-primary"
        disabled={true}
        onClick={[Function]}
        type="button"
      >
        Add Contact
      </button>
    </div>
  </div>,
]
`;
